plugins/plugins/api/api.js

```diff
@@ -36,12 +38,12 @@ var plugin = {},
                     plugins.dispatch("/systemlogs", {params: params, action: "change_plugins", data: {before: before, update: params.qstring.plugin}});
                     process.send({ cmd: "startPlugins" });
                     plugins.syncPlugins(params.qstring.plugin, function(err) {
-                        process.send({ cmd: "endPlugins" });
-                        if (err) {
-                            common.returnOutput(params, 'Errors');
+                        if (!err) {
+                            process.send({ cmd: "endPlugins" });
+                            updatePluginState("end");
                         }
                         else {
-                            common.returnOutput(params, 'Success');
+                            common.returnMessage(params, 500, "failed");
```

Comment:
I don't think a (failed) response will be sent from here because you have already sent the (started) message in line no. 27